File: wrappers/dotnet/src/Hyperledger.Ursa.BbsSignatures/Models/VerifyBlindedCommitmentRequest.cs
using System;

namespace Hyperledger.Ursa.BbsSignatures
{
    public class VerifyBlindedCommitmentRequest
    {
        public VerifyBlindedCommitmentRequest(BbsKeyPair publicKey, byte[] proof, uint[] blindedIndices, string nonce)
        {
            PublicKey = publicKey ?? throw new ArgumentNullException(nameof(publicKey));
            Proof = proof ?? throw new ArgumentNullException(nameof(proof));
            BlindedIndices = blindedIndices ?? throw new ArgumentNullException(nameof(blindedIndices));
            Nonce = nonce ?? throw new ArgumentNullException(nameof(nonce));
        }

        public BbsKeyPair PublicKey { get; set; }

        public byte[] Proof { get; set; }

        public uint[] BlindedIndices { get; set; }

        public string Nonce { get; set; }
    }
}

File: wrappers/python/ursa_bbs_signatures/models/VerifyBlindedCommitmentRequest.py
from typing import List
from .IndexedMessage import IndexedMessage
from .keys import BbsKey


class VerifyBlindedCommitmentRequest:
    def __init__(
        self,
        public_key: BbsKey,
        proof: bytes,
        blinded_indices: List[IndexedMessage],
        nonce: bytes,
    ) -> None:
        """ Verify Blinded Commitment Request

        Args:
            public_key: The pubic key
            proof: The proof
            blinded_indices: The blinded indices
            nonce: The nonce
        """
        self.public_key = public_key
        self.proof = proof
        self.blinded_indices = blinded_indices
        self.nonce = nonce
